python/ray/serve/_private/config.py

```diff
@@ -162,6 +82,12 @@ def to_proto(self):
         data["user_configured_option_names"] = list(
             data["user_configured_option_names"]
         )
+        # TODO(zcin): This is temporary before these options are moved
+        # from ReplicaConfig to DeploymentConfig
```

Comment:
Don't totally understand this TODO -- I think the split between replica config & deployment config right now is useful (things that are dynamically updateable vs. things that aren't).

Are you planning to unify them?

Reply:
The external pydantic model groups everything together (e.g. [num replicas, graceful shutdown etc] + [ray actor options, placement group etc]), so I also have all these options together in `BaseDeploymentModel`. I was planning on putting everything that users can configure through the Serve config in `BaseDeploymentModel`, and only (deployment def, init args, init kwargs) in `ReplicaInitInfo`. (This would also make the logic in [version.py](https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/version.py#L84-L119) much simpler)

I am also open to keeping the separation that we have right now, would just have to prototype restructure the pydantic models since the goal is to only have one unified location for each option (right now it's duplicated in `BaseDeploymentModel` and `ReplicaInitInfo`)

Reply:
Discussed offline - I'll be sticking with the new split between `InternalDeploymentConfig` and `ReplicaInitInfo`.